### PR TITLE
ADEN-5992 Fixed ads-app-no-ads-for-users-oasis job after PAL enabled in US

### PR DESCRIPTION
**Pull request by nandy-andy**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L362](https://github.com/Wikia/selenium-tests/blob/7e8de247fd9bc0d96769993f1ca7794218cd6086/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L362): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L442](https://github.com/Wikia/selenium-tests/blob/7e8de247fd9bc0d96769993f1ca7794218cd6086/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L442): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 1]
```diff
@@ -154,8 +154,7 @@ public void verifyTopLeaderboard() {
   }
 
   public void verifyNoAdsOnPage() {
-    scrollToSelector(AdsContent.getSlotSelector(AdsContent.ADS_IN_CONTENT_CONTAINER));
-    scrollToSelector(AdsContent.getSlotSelector(AdsContent.PREFOOTERS_CONTAINER));
+    scrollToSelector(AdsContent.getSlotSelector(AdsContent.BOTTOM_LB));
```
Comment (fraszczakszymon) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: Shouldn't be a part of logic in verifyNoAd?

Can we reuse SLOT_TRIGGERS in verifyNoAd and scroll to given slot when it's necessary?

https://github.com/Wikia/selenium-tests/blob/7e8de247fd9bc0d96769993f1ca7794218cd6086/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java#L96

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L362](https://github.com/Wikia/selenium-tests/blob/ac451fa044d94175c3569f74249c8c2994470eed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L362): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L442](https://github.com/Wikia/selenium-tests/blob/ac451fa044d94175c3569f74249c8c2994470eed/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L442): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L369](https://github.com/Wikia/selenium-tests/blob/52dfe2af6e78bb86e6975f52fa8838cdb108a041/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L369): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L449](https://github.com/Wikia/selenium-tests/blob/52dfe2af6e78bb86e6975f52fa8838cdb108a041/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L449): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L369](https://github.com/Wikia/selenium-tests/blob/9ee1a9a619a823744f8a1d86a4d5f0b1753cdea1/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L369): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L449](https://github.com/Wikia/selenium-tests/blob/9ee1a9a619a823744f8a1d86a4d5f0b1753cdea1/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L449): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L369](https://github.com/Wikia/selenium-tests/blob/30c1718d5ccdfcb45bfb7af8000ee08c795e03e2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L369): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L449](https://github.com/Wikia/selenium-tests/blob/30c1718d5ccdfcb45bfb7af8000ee08c795e03e2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L449): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 2]
```diff
@@ -664,9 +676,17 @@ private void verifySlotExpanded(WebElement element) {
...
-      verifyNoAd(selector);
+    Map<String, String> slots = AdsContent.getSlotsSelectorsMap();
+    for (Map.Entry<String, String> entry : slots.entrySet()) {
+      final String slotName = entry.getKey();
+      final String selector = entry.getValue();
+      final String javaScriptTrigger = AdsContent.getSlotTrigger(slotName);
+
+      if (StringUtils.isEmpty(javaScriptTrigger)) {
+        verifyNoAd(selector);
+      } else {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: MAJOR Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L369](https://github.com/Wikia/selenium-tests/blob/1794ceb86d399c1a2e6131834bb456cee731bf4a/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L369): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AdsBaseObject.java#L449](https://github.com/Wikia/selenium-tests/blob/1794ceb86d399c1a2e6131834bb456cee731bf4a/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L449): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 3]
```diff
@@ -433,14 +442,21 @@ public boolean checkSlotOnPageLoaded(String slotName) {
     WebElement slot;
     changeImplicitWait(250, TimeUnit.MILLISECONDS);
     try {
+      String slotSelector = AdsContent.getSlotSelector(slotName);
+
       if (slotName.equals(AdsContent.FLOATING_MEDREC)) {
-        triggerAdSlot(AdsContent.FLOATING_MEDREC);
+        triggerAdSlot(AdsContent.FLOATING_MEDREC).
+            wait.
+            forElementPresent(By.cssSelector(slotSelector));
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: this \n looks strange here. If we want to split into multiple lines let's put `.` at the beginning of the line

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotSizes.java [hunk 4]
```diff
@@ -67,8 +67,11 @@ private void adsSlotSizes(Page page,
       // only desktop pages have this headline
       ads.scrollToPosition(ARTICLE_MIDDLE_SECTION_SELECTOR);
     }
-    ads.wait.forElementPresent(By.cssSelector(AdsContent.getSlotSelector(slotName)));
-    ads.triggerAdSlot(slotName);
+
+    ads.triggerAdSlot(slotName).
+        wait.
+        forElementPresent(By.cssSelector(AdsContent.getSlotSelector(slotName)));
+
```
Comment (dianafa) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotSizes.java]: same here with `.` and `\n` please :))

**Comment by dianafa** [comment]
LGTM

**Pull request merged**